libsmtutil/SMTPortfolio.cpp

```diff
@@ -36,11 +36,13 @@ SMTPortfolio::SMTPortfolio(
 	map<h256, string> _smtlib2Responses,
 	frontend::ReadCallback::Callback _smtCallback,
 	[[maybe_unused]] SMTSolverChoice _enabledSolvers,
-	optional<unsigned> _queryTimeout
+	optional<unsigned> _queryTimeout,
+	bool _printSmt
 ):
 	SolverInterface(_queryTimeout)
 {
-	if (_enabledSolvers.smtlib2)
+	smtlib2Enabled = _enabledSolvers.smtlib2;
+	if (_enabledSolvers.smtlib2 || _printSmt)
```

Comment:
We could also require that the user must enable `smtlib2` as a solver if they want to print the query, it would simplify this part. What do you think?

Reply:
I agree.